packages/story-editor/src/components/library/panes/media/common/insertionMenu.js

```diff
@@ -0,0 +1,197 @@
+ * @param {Object} props.resource Selected media element's resource object.
+ * @param {boolean} props.display Whether the more icon should be displayed.
+ * @param {Function} props.onInsert Callback for inserting media.
+ * @param {number} props.width Media width.
+ * @param {number} props.index Element index in the gallery.
+ * @param {boolean} props.isLocal If the menu is for local or 3p media.
+ * @return {null|*} Element or null if should not display the More icon.
+ */
+function InsertionMenu({ resource, display, onInsert, width, index, isLocal }) {
+  const insertButtonRef = useRef();
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+  const onMenuOpen = useCallback((e) => {
+    e.stopPropagation();
+    setIsMenuOpen(true);
+  }, []);
+  const onMenuCancelled = useCallback(() => setIsMenuOpen(false), []);
+
+  const { currentBackgroundId, combineElements } = useStory((state) => ({
+    currentBackgroundId: state.state.currentPage?.elements?.[0]?.id,
+    combineElements: state.actions.combineElements,
+  }));
+
+  const { type, poster } = resource;
+  const insertLabel = ['image', 'gif'].includes(type)
+    ? __('Insert image', 'web-stories')
+    : __('Insert video', 'web-stories');
+  const options = [
+    {
+      group: [
+        { label: insertLabel, value: MENU_OPTIONS.INSERT },
+        {
+          label: __('Add as background', 'web-stories'),
+          value: MENU_OPTIONS.ADD_BACKGROUND,
+        },
+      ],
+    },
+  ];
+
+  const handleCurrentValue = (evt, value) => {
+    const thumbnailUrl = poster || getSmallestUrlForWidth(width, resource);
+    const newElement = getElementProperties(resource.type, {
+      resource,
+    });
+    onMenuCancelled();
+    switch (value) {
+      case MENU_OPTIONS.INSERT:
+        onInsert(resource, thumbnailUrl);
+        break;
+      case MENU_OPTIONS.ADD_BACKGROUND:
+        combineElements({
+          firstElement: newElement,
+          secondId: currentBackgroundId,
+        });
+        break;
+      default:
+        break;
+    }
+  };
+
+  useRovingTabIndex({ ref: insertButtonRef }, [], BUTTON_NESTING_DEPTH);
+  // In case of 3p media, we tab out of the panel since it's the only button there.
+  const focusCanvas = useFocusCanvas();
+  useKeyDownEffect(isLocal ? null : insertButtonRef, 'tab', focusCanvas, [
+    focusCanvas,
+  ]);
+
+  const listId = useMemo(() => `list-${uuidv4()}`, []);
+  const buttonId = useMemo(() => `button-${uuidv4()}`, []);
+
+  // Keep icon and menu displayed if menu is open (even if user's mouse leaves the area).
+  return (
+    <MenuContainer>
+      <ActionButton
+        ref={insertButtonRef}
+        onClick={onMenuOpen}
+        aria-label={__('Open insertion menu', 'web-stories')}
+        aria-pressed={isMenuOpen}
+        aria-haspopup
+        aria-expanded={isMenuOpen}
+        aria-owns={isMenuOpen ? listId : null}
+        id={buttonId}
+        $display={display}
+        tabIndex={index === 0 ? 0 : -1}
+      >
```

Comment:
If you tab out of the media item, while the insertion menu is open, this action button stays visible. This also means, that if you shift-tab back to the media item, this action button receives keyboard focus, but you cannot use arrow keys to navigate to neighboring media items.

How to reproduce:

* Move keyboard focus to local media item
* Press enter to open insertion menu
* Press tab to move focus to canvas
* Press shift-tab to move focus back to media item, and observe focus moves to insertion menu button
* Try to navigate to other media items using arrow keys and observe it **does not** work
* Press shift-tab again to move focus back to the entire media item
* Try to navigate to other media items using arrow keys and observe it **does** work

I think the solution is to always hide this button once the media item loses `focus-within`, but not sure.

![insertion-tab-bug](https://user-images.githubusercontent.com/637548/154754788-68b8f70c-ddd7-4653-bac1-7fbff5d6836a.gif)

Reply:
Looks like the same behaviour is happening on main as well. Included this into https://github.com/GoogleForCreators/web-stories-wp/issues/10607